mk/tests.mk

```diff
@@ -0,0 +1,49 @@
+CACHE_TEST_DIR := tests/cache
+CACHE_BUILD_DIR := build/cache
+TARGET := test_cache
+
+CACHE_OBJS := cache.o \
+        test_cache.o
+
+CACHE_OBJS := $(addprefix $(CACHE_BUILD_DIR)/, $(CACHE_OBJS))
+OBJS += $(CACHE_OBJS)
+deps += $(CACHE_OBJS:%.o=%.o.d)
+
+CACHE_CHECK_ELF_FILES := \
+	cache_new \
+	cache_put \
+	cache_get \
+	cache_lru_replace \
+	cache_lfu_replace \
+	cache_lfu_ghost_replace \
+
+CACHE_OUT = $(addprefix $(CACHE_BUILD_DIR)/, $(CACHE_CHECK_ELF_FILES:%=%.out))
+
+tests : $(CACHE_OUT)
+	$(Q)$(foreach e,$(CACHE_CHECK_ELF_FILES),\
+		$(PRINTF) "Running $(e) ... "; \
+		if cmp $(CACHE_TEST_DIR)/$(e).expect $(CACHE_BUILD_DIR)/$(e).out; then \
+		$(call notice, [OK]); \
+		else \
+		$(PRINTF) "Failed.\n"; \
+		exit 1; \
+		fi; \
+	)
+
+$(CACHE_OUT): $(TARGET)
+	$(Q)$(foreach e,$(CACHE_CHECK_ELF_FILES),\
+		$(CACHE_BUILD_DIR)/$(TARGET) $(CACHE_TEST_DIR)/$(e).in > $(CACHE_BUILD_DIR)/$(e).out; \
+	)
+
+$(TARGET): $(CACHE_OBJS)
+	$(VECHO) "  CC\t$@\n"
+	$(Q)$(CC) $^ -o $(CACHE_BUILD_DIR)/$(TARGET)
+	
+$(CACHE_BUILD_DIR)/cache.o: src/cache.c 
```

Comment:
You can unify the rules by the following:
```
$(CACHE_BUILD_DIR)/%.o: $(CACHE_TEST_DIR)/%.c
```

Reply:
The directory of `cache.c` is `src`, but the directory of `test_cache.c` is `tests/cache`. So I cannot nuify the rules to this.

Reply:
No, you don't have to build `$(CACHE_BUILD_DIR)/cache.o`. Instead, only `$(CACHE_BUILD_DIR)/test_cache.o` should be built here. The test oriented rules always depend on the generation of `$(OBJS)`.

Reply:
I see, `cache.o` would be built by `make all`

Reply:
However, the GitHub Actions would fail now, because we don't integrate the cache and it would not builf `cache.o`.

Reply:
The appropriate sequence would be
```
make
make tests
```

Alternatively, you can extend item `check` to perform `make tests`.

Reply:
By the way, the next tests would be associated with #94 .